metasrv/tests/it/api/http/clsuter_state_test.rs

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2021 Datafuse Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+use common_base::tokio;
+use databend_meta::api::http::v1::cluster_state::cluster_node_handler;
+use databend_meta::api::http::v1::cluster_state::cluster_state_handler;
+use poem::get;
+use poem::http::Method;
+use poem::http::StatusCode;
+use poem::http::Uri;
+use poem::Endpoint;
+use poem::Request;
+use poem::Route;
+use pretty_assertions::assert_eq;
+
+#[tokio::test]
+async fn test_cluster_nodes() -> common_exception::Result<()> {
+    let (_log_guards, ut_span) = init_meta_ut!();
+    let _ent = ut_span.enter();
+    let tc = new_test_context(0);
+    let meta_node = MetaNode::open_create_boot(&tc.config.raft_config, None, Some(()), None).await?;
+
+    let cluster_router = Route::new().at("/cluster/nodes").get(get!(health_handler)).data(meta_node.clone());
+    let response = cluster_router.call(Request::new(Method::Get, Uri::from_static("/cluster/nodes"))).await?;
+    assert_eq!(response.status(), StatusCode::OK);
+    meta_node.stop().await?;
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_cluster_state() -> common_exception::Result<()> {
+    let (_log_guards, ut_span) = init_meta_ut!();
+    let _ent = ut_span.enter();
+    let tc = new_test_context(0);
+    let meta_node = MetaNode::open_create_boot(&tc.config.raft_config, None, Some(()), None).await?;
+
+    let cluster_router = Route::new().at("/cluster/nodes").get(get!(health_handler)).data(meta_node.clone());
+    let response = cluster_router.call(Request::new(Method::Get, Uri::from_static("/cluster/nodes"))).await?;
+    assert_eq!(response.status(), StatusCode::OK);
+    meta_node.stop().await?;
+    Ok(())
```

Comment:
Is this test faking a http-service?

I supposed the test should test the service defined in the code. 🤔 
E.g., start a real metasrv(not just meta-node), and send an http request to it.
Then check the response as expected.

To bring a cluster with more than one servers, this is a good reference: 
https://github.com/datafuselabs/databend/blob/46a2e816037c6d76991cec64db70b86cd20d5b83/metasrv/tests/it/flight/metasrv_flight_api.rs#L131-L146

Reply:
It's not a faking http-services now. 
we can test the response with multiple meta-node, no need to start a real metasrv.